### PR TITLE
Add manual impression tracking to select ad formats

### PR DESCRIPTION
**Pull request by chauduyphanvu1**
*AdColony*
Interstitial

*AdMob*
Banner
Interstitial

*AppLovin*
Banner
Interstitial

*Chartboost*
Interstitial

*Facebook*
Banner
Interstitial

*Flurry*
Interstitial

*ironSource*
Interstitial

*One by AOL / Millennial*
-Banner- SDK does not have an equivalent callback for impression tracking / ad opened: https://github.com/mopub/mopub-android-mediation/blob/master/OnebyAOL/com/mopub/mobileads/MillennialBanner.java#L164-L270
Interstitial

*Tapjoy*
Interstitial

*Unity*
Interstitial

*Vungle*
Interstitial

### CODE REVIEW

**Comment by wacamoe** [comment]
Can we confirm that these same networks are the same ones currenly using the feature on iOS? We want to ensure we aren't adding it to ones that iOS isn't also using these impression callbacks.

**Comment by chauduyphanvu1** [comment]
@wacamoe - on it! 

We could also end up doing the reverse - having more iOS adapters implement manual imp tracking, since more network SDKs will likely have updated and exposed dedicated imp callbacks, too. This list for Android is rather long but there are more network SDKs there that can do it.

File: AdColony/com/mopub/mobileads/AdColonyInterstitial.java [hunk 1]
```diff
@@ -179,6 +181,7 @@ public void onOpened(@NonNull com.adcolony.sdk.AdColonyInterstitial ad) {
                         @Override
                         public void run() {
                             mCustomEventInterstitialListener.onInterstitialShown();
+                            mCustomEventInterstitialListener.onInterstitialImpression();
```
Comment (bichenwang) [inline AdColony/com/mopub/mobileads/AdColonyInterstitial.java]: this actually does nothing different. onInterstitialShown() does the same exact thing as onInterstitialImpression when automatic impression and click tracking is false.
So this change is exactly the same as having automatic impression true. So let's try to limit this to networks that have this on iOS.

File: AdMob/com/mopub/mobileads/GooglePlayServicesBanner.java [hunk 2]
```diff
@@ -127,6 +130,12 @@ private AdSize calculateAdSize(int width, int height) {
         /*
          * Google Play Services AdListener implementation
          */
+
+        @Override
+        public void onAdImpression() {
+            mBannerListener.onBannerImpression();
```
Comment (bichenwang) [inline AdMob/com/mopub/mobileads/GooglePlayServicesBanner.java]: would recommend a null check on all these onBannerImpression calls

**Pull request closed without merging**